product-api/data/products.go

```diff
@@ -92,7 +92,10 @@ func DeleteProduct(id int) error {
 		return ErrProductNotFound
 	}
 
-	productList = append(productList[:i], productList[i+1])
+	copy(productList[i:], productList[i+1:])
```

Comment:
Hey, I think you could write it like `productList = append(productList[:i], productList[i+1:]...)` and it should work as well. I've tested it and got the right product deleted.